daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -40,133 +41,180 @@ import scalaz.{@@, Tag}
 
 private[lf] object Anf {
```

Comment:
Note to reviewers: the only interesting change in this file is the addition of `transformMultiAppRightToLeft`. The rest of the diff is noise due to moving some methods around.

I had to moved these methods around because I turned `Anf` into a class in order to parameterize it by the evaluation mode. Doing that forced me to move all static methods together into the object and all non-static ones together into the class. Thus most of the diff is methods being moved to another location